gems/smithy/lib/smithy/anvil/client/views/endpoint_provider.rb

```diff
@@ -0,0 +1,244 @@
+# frozen_string_literal: true
+
+require 'stringio'
+
+module Smithy
+  module Anvil
+    module Client
+      module Views
+        # @api private
+        class EndpointProvider < View
+          def initialize(plan)
+            @plan = plan
+            @model = plan.model
+            service = @plan.service.values.first
```

Comment:
I know this is a sharp edge. One other way is `_, service = @plan.service.first`